src/configuration.rs

```diff
@@ -89,5 +100,31 @@ pub trait Operation: Send + Sync {
     /// Returns ControlFlow::Break if it should finish work, for example
     /// if the operation ID has exceeded the configured operation count.
     /// In other cases, it returns ControlFlow::Continue.
-    async fn execute(&mut self, ctx: &OperationContext) -> Result<ControlFlow<()>>;
+    async fn run(&mut self, session: WorkerSession) -> Result<()>;
 }
+
+/// Implements Operation for a type which implements an execute method.
+/// Although we could put execute() into the Operation trait, doing what we
+/// are doing here has better performance because asynchronous traits require
+/// putting returned futures in a Box due to current language limitations.
+/// Boxing the futures imply an allocation per operation and those allocations
+/// can be clearly visible on the flamegraphs.
+#[macro_export]
+macro_rules! make_runnable {
```

Comment:
I think this would be prettier as a `#[derive(Runnable)]` macro.
That would require more work, so I guess this can stay if there's no time to do it.

Reply:
I've already done that in another PR on top of this: #35 !

Reply:
I'm sorry, I changed my mind about the procedural macro. Procedural macros need to be put in a separate crate and when we will publish cql-stress to crates.io we will have to publish the new crate, too - which will make maintenance harder. I don't think converting to a derive macro is worth it.